atddiff/src/bin/Atddiff_main.ml

```diff
@@ -0,0 +1,201 @@
+(*
+   Entry point to the atddiff command.
+*)
+
+open Printf
+open Cmdliner
+
+type conf = {
+  old_file: string;
+  new_file: string;
+  out_file: string option;
+  json_defaults_old: bool;
+  json_defaults_new: bool;
+  version: bool;
+}
+
+let run conf =
+  if conf.version then (
+    print_endline Atddiff.version;
+    exit 0
+  )
+  else
+    let out_data =
+      Atddiff.compare_files
+        ~json_defaults_old:conf.json_defaults_old
+        ~json_defaults_new:conf.json_defaults_new
+        conf.old_file conf.new_file in
+    match conf.out_file with
+    | None -> print_string out_data
+    | Some out_file ->
+       let oc = open_out_bin out_file in
+       Fun.protect
+         ~finally:(fun () -> close_out_noerr oc)
+         (fun () -> output_string oc out_data)
+
+(***************************************************************************)
+(* Command-line processing *)
+(***************************************************************************)
+
+let error msg =
+  eprintf "Error: %s\n%!" msg;
+  exit 1
+
+let old_file_term =
+  let info =
+    Arg.info []
+      ~docv:"OLD_ATD_FILE"
+      ~doc:"Path to the older version of the ATD file to compare"
+  in
+  Arg.required (Arg.pos 0 (Arg.some Arg.file) None info)
+
+let new_file_term =
+  let info =
+    Arg.info []
+      ~docv:"NEW_ATD_FILE"
+      ~doc:"Path to the newer version of the ATD file to compare"
+  in
+  Arg.required (Arg.pos 1 (Arg.some Arg.file) None info)
+
+let out_file_term =
+  let info =
+    Arg.info ["o"; "output-file"]
+      ~docv:"OUTPUT_FILE"
+      ~doc:"Path to the output file. The default is to print the result to \
+            standard output."
+  in
+  Arg.value (Arg.opt (Arg.some Arg.string) None info)
+
+let json_defaults_term =
+  let info =
+    Arg.info ["json-defaults"]
+      ~doc:"Shorthand for '--json-defaults-old --json-defaults-new'."
+  in
+  Arg.value (Arg.flag info)
+
+let json_defaults_old_term =
```

Comment:
I like to put the type on those things, like let json_defaults_old_term : bool Term.t = ...
It helps a bit.

Reply:
I did what you suggest. I like it.